File: back/src/models/Makhdoom.js
import mongoose from 'mongoose';

const makhdoomSchema = new mongoose.Schema(
  {
    id: { type: String, index: true },
    name: { type: String, required: true, trim: true, index: true },
    gender: { type: String, default: 'male' },
    osra: { type: String, default: '', index: true },
    stage: { type: String, default: '', index: true },
    fasl: { type: String, default: '', index: true },
    phone: { type: String, default: '', trim: true },
    phoneNumber: { type: String, default: '', trim: true },
    address: { type: String, default: '' },
    area: { type: String, default: '' },
    street: { type: String, default: '' },
    building: { type: String, default: '' },
    floor: { type: String, default: '' },
    apartment: { type: String, default: '' },
    notes: { type: String, default: '' },
    serviceYear: { type: String, default: '2026', index: true },
    status: { type: String, default: 'active', index: true },
    assignedServant: { type: String, default: null, index: true },
    code: { type: String, default: '' },
    timestamp: { type: String, default: () => new Date().toISOString() },
  },
  {
    timestamps: true,
    strict: false,
    collection: 'makhdomeen',
  }
);

makhdoomSchema.index({ serviceYear: 1, osra: 1, stage: 1, fasl: 1 });
makhdoomSchema.index({ assignedServant: 1, serviceYear: 1 });

const Makhdoom = mongoose.models.Makhdoom || mongoose.model('Makhdoom', makhdoomSchema);
export default Makhdoom;

File: back/Models/Makhdoom.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Eftekad.Backend.Models
{
    [Table("makhdomeen")]
    public class Makhdoom
    {
        [Key]
        [Column("id")]
        public string Id { get; set; } = string.Empty;

        [Column("name")]
        public string Name { get; set; } = string.Empty;

        [Column("gender")]
        public string Gender { get; set; } = "male";

        [Column("osra")]
        public string Osra { get; set; } = string.Empty;

        [Column("stage")]
        public string Stage { get; set; } = string.Empty;

        [Column("fasl")]
        public string Fasl { get; set; } = string.Empty;

        [Column("phone")]
        public string Phone { get; set; } = string.Empty;

        [Column("address")]
        public string Address { get; set; } = string.Empty;

        [Column("area")]
        public string Area { get; set; } = string.Empty;

        [Column("street")]
        public string Street { get; set; } = string.Empty;

        [Column("building")]
        public string Building { get; set; } = string.Empty;

        [Column("floor")]
        public string Floor { get; set; } = string.Empty;

        [Column("apartment")]
        public string Apartment { get; set; } = string.Empty;

        [Column("notes")]
        public string Notes { get; set; } = string.Empty;

        [Column("serviceYear")]
        public string ServiceYear { get; set; } = "2026";

        [Column("status")]
        public string Status { get; set; } = "active";

        [Column("assignedServant")]
        public string? AssignedServant { get; set; }

        [Column("code")]
        public string Code { get; set; } = string.Empty;

        [Column("timestamp")]
        public string Timestamp { get; set; } = DateTime.UtcNow.ToString("o");

        [Column("birthDate")]
        public string BirthDate { get; set; } = string.Empty;

        [Column("pendingPromotionFrom")]
        public string? PendingPromotionFrom { get; set; }

        [Column("createdAt")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [Column("updatedAt")]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}
